python/src/iceberg/utils/datetime.py

```diff
@@ -63,3 +65,27 @@ def timestamptz_to_micros(timestamptz_str: str) -> int:
     if ISO_TIMESTAMPTZ.fullmatch(timestamptz_str):
         return datetime_to_micros(datetime.fromisoformat(timestamptz_str))
     raise ValueError(f"Invalid timestamp with zone: {timestamptz_str} (must be ISO-8601)")
+
+
+def to_human_day(day_ordinal: int) -> str:
+    """Converts a DateType value to human string"""
+    to_time = EPOCH_TIMESTAMP + timedelta(days=day_ordinal)
+    return "{0:0=4d}-{1:0=2d}-{2:0=2d}".format(to_time.year, to_time.month, to_time.day)
+
+
+def to_human_time(micros_from_midnight: int) -> str:
+    """Converts a TimeType value to human string"""
+    to_day = EPOCH_TIMESTAMP + timedelta(microseconds=micros_from_midnight)
```

Comment:
Time shouldn't use epoch. There's no need to mix dates into this.

Getting a time is actually straightforward:

```python
def time_from_micros(micros: int) -> time:
    seconds = micros // 1_000_000
    minutes = seconds // 60
    hours = minutes // 60
    time(hour=hours, minute=minutes % 60, second=seconds % 60, microsecond=micros % 1_000_000)
```

Reply:
This should use the code provided here, rather than creating a datetime and converting it to a time.

Reply:
👌  Updated.